pages/chat.html

```diff
@@ -0,0 +1,194 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <title>Fastify WebSockets chat</title>
+  <meta charset="utf-8" />
+  <meta name="viewport" content="initial-scale=1, width=device-width" />
+  <script src="https://unpkg.com/react@latest/umd/react.development.js" crossorigin="anonymous"></script>
+  <script src="https://unpkg.com/react-dom@latest/umd/react-dom.development.js"></script>
+  <script src="https://unpkg.com/@mui/material@latest/umd/material-ui.development.js" crossorigin="anonymous"></script>
+  <script src="https://unpkg.com/babel-standalone@latest/babel.min.js" crossorigin="anonymous"></script>
+  <!-- Fonts to support Material Design -->
+  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
+  <!-- Icons to support Material Design -->
+  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
+</head>
+
+<body>
+  <div id="root"></div>
+  <script type="text/babel">
+    /**
+     * PLEASE DON'T BLAME ME FOR THIS ONE SINGLE HUGE FILE.
+     * The target of this file is for the purpose of learning WebSocket.
+     * It's not meant to be a production-ready solution.
+     * It's just a proof of concept.
+     * 
+     * Thank you for your understanding.
+     */
+
+    const {
+      colors,
+      CssBaseline,
+      ThemeProvider,
+      Typography,
+      TextField,
+      TextareaAutosize,
+      Button,
+      Grid,
+      Container,
+      createTheme,
+      Box,
+      SvgIcon,
+      Link,
+    } = MaterialUI;
+
+    // Create a theme instance.
+    const theme = createTheme({
+      palette: {
+        primary: { main: '#556cd6', },
+        secondary: { main: '#19857b', },
+        error: { main: colors.red.A400, },
+      },
+    });
+
+    function WsConnection(prop) {
+      const [ws, setWs] = React.useState(null);
```

Comment:
Looks like you're not using `ws` anywhere, I suggest to remove this state. If you need to keep a reference to the websocket instance you may want to use `useRef` instead.